pvlib/modelchain.py

```diff
@@ -935,13 +935,18 @@ def losses_model(self, model):
         elif isinstance(model, str):
             model = model.lower()
             if model == 'pvwatts':
-                self._losses_model = self.pvwatts_losses
+                self._losses_model = model
+                self._dc_losses = self.pvwatts_losses
             elif model == 'no_loss':
-                self._losses_model = self.no_extra_losses
+                self._losses_model = model
+                self._dc_losses = self.no_extra_losses
+            elif model == 'pvsyst':
+                self._losses_model = model
+                self._dc_losses = self.pvsyst_dc_losses
             else:
                 raise ValueError(model + ' is not a valid losses model')
         else:
-            self._losses_model = partial(model, self)
```

Comment:
-1 on this change. It's a powerful feature (albeit likely seldom-used) and the change would make `losses_model` inconsistent with the other model options.

Reply:
I went back a forth about how to set this up, and agreed that this makes it inconsistent. How about treating `losses_model` somewhat like `clearsky_model` or `transposition_model` where it is always just a string and then that string can trigger different behavior at the various losses steps in the chain? The difference there though is that those tags are strings because they get passed to other functions that way, and losses_model would behave differently if we did it like that. I liked the functionality of the setter as a way both to initialize appropriate functions and work as a check for a compatible model. 

But, we could set it up that if `losses_model` does not have a setter and is a string, there is a `ModelChain._dc_losses` function that runs, and that calls different functions (`pvwatts`, `pvsyst_dc_losses`, `no_extra_losses`, etc) that get triggered based on what string `ModelChain.losses_model` is. And similarly for other loss functions like a `ModelChain._shading_losses`.

Reply:
The more I think about it, the more I do lean towards having each loss (snow, shading, ohmic, etc) be its own `ModelChain` kwarg. It increases the amount of setup a user has to do, because they have to pass in each of those if they want them, instead of just say passing in "pvsyst", or "pvwatts" and loss models, but it would allow a user to mix and match methods which is how all the rest of the methods are setup. Then we can add in helpers kind of like the `with_sapm` one that can setup default configurations. 

This way will add in more code overall, I think, but it will keep it consistent an flexible. I will work to build that out a bit and then submit it for review.